### ISSUE TITLE
sass exceptions break guard-compass until guard restarted

### ISSUE THREAD

**Issue by aaronmcadam** [guard/guard-compass#6]
Similar to this bug in guard-sass: https://github.com/hawx/guard-sass/issues/15

Seems to be down to specific gem versions

**Comment by aaronmcadam** [comment]
Stack trace here: https://gist.github.com/1386047

**Comment by narkeeso** [comment]
Supposedly issue #7 was suppose to fix this. Would be nice if this could be fixed.

Edit: nevermind, seems like rubygems is using an older version of this. Add this to your bundle:

gem "guard-compass", :git =>
    "https://github.com/guard/guard-compass.git"

**Comment by aaronmcadam** [comment]
oh, thanks! I'll try this later :)

**Comment by ghost** [comment]
I'm having this issue as well.

```
23:37:34 - ERROR - Guard::Compass failed to achieve its <run_on_change>, exception was:
> [#] Sass::SyntaxError: Invalid CSS after "    padding-bottom": expected ";", was ": 0.5em;"
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/scss/parser.rb:1148:in `expected'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/scss/parser.rb:1084:in `expected'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/scss/parser.rb:1079:in `tok!'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/scss/parser.rb:586:in `block in declaration_or_ruleset'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/scss/parser.rb:1122:in `call'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/scss/parser.rb:1122:in `rethrow'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/scss/parser.rb:592:in `declaration_or_ruleset'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/scss/parser.rb:554:in `block_child'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/scss/parser.rb:543:in `block_contents'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/scss/parser.rb:535:in `block'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/scss/parser.rb:529:in `ruleset'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/scss/parser.rb:553:in `block_child'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/scss/parser.rb:546:in `block_contents'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/scss/parser.rb:82:in `stylesheet'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/scss/parser.rb:27:in `parse'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/engine.rb:342:in `_to_tree'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/engine.rb:274:in `to_tree'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/tree/visitors/perform.rb:225:in `visit_import'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/tree/visitors/base.rb:37:in `visit'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/tree/visitors/perform.rb:100:in `visit'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/tree/visitors/base.rb:53:in `block in visit_children'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/tree/visitors/base.rb:53:in `map'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/tree/visitors/base.rb:53:in `visit_children'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/tree/visitors/perform.rb:109:in `block in visit_children'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/tree/visitors/perform.rb:121:in `with_environment'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/tree/visitors/perform.rb:108:in `visit_children'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/tree/visitors/base.rb:37:in `block in visit'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/tree/visitors/perform.rb:128:in `visit_root'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/tree/visitors/base.rb:37:in `visit'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/tree/visitors/perform.rb:100:in `visit'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/tree/visitors/perform.rb:7:in `visit'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/tree/root_node.rb:20:in `render'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/engine.rb:315:in `_render'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/sass-3.2.7/lib/sass/engine.rb:262:in `render'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/compass-0.12.2/lib/compass/compiler.rb:140:in `block (2 levels) in compile'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/compass-0.12.2/lib/compass/compiler.rb:126:in `timed'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/compass-0.12.2/lib/compass/compiler.rb:139:in `block in compile'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/compass-0.12.2/lib/compass/logger.rb:45:in `red'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/compass-0.12.2/lib/compass/compiler.rb:138:in `compile'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/compass-0.12.2/lib/compass/compiler.rb:118:in `compile_if_required'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/compass-0.12.2/lib/compass/compiler.rb:103:in `block (2 levels) in run'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/compass-0.12.2/lib/compass/compiler.rb:101:in `each'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/compass-0.12.2/lib/compass/compiler.rb:101:in `block in run'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/compass-0.12.2/lib/compass/compiler.rb:126:in `timed'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/compass-0.12.2/lib/compass/compiler.rb:100:in `run'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/compass-0.12.2/lib/compass/commands/update_project.rb:45:in `perform'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/compass-0.12.2/lib/compass/commands/base.rb:18:in `execute'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/compass-0.12.2/lib/compass/commands/project_base.rb:19:in `execute'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-compass-0.0.6/lib/guard/compass.rb:47:in `run_on_change'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard/runner.rb:99:in `block in run_supervised_task'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard/runner.rb:97:in `catch'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard/runner.rb:97:in `run_supervised_task'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard/runner.rb:146:in `block in run_first_task_found'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard/runner.rb:144:in `each'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard/runner.rb:144:in `run_first_task_found'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard/runner.rb:79:in `block in run_on_changes'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard/runner.rb:175:in `block (3 levels) in scoped_guards'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard/runner.rb:174:in `each'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard/runner.rb:174:in `block (2 levels) in scoped_guards'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard/runner.rb:173:in `catch'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard/runner.rb:173:in `block in scoped_guards'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard/runner.rb:172:in `each'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard/runner.rb:172:in `scoped_guards'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard/runner.rb:72:in `run_on_changes'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard.rb:141:in `block (2 levels) in setup_listener'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard.rb:365:in `block in within_preserved_state'
> [#] <internal:prelude>:10:in `synchronize'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard.rb:362:in `within_preserved_state'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/guard-1.7.0/lib/guard.rb:140:in `block in setup_listener'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/listen-0.7.3/lib/listen/listener.rb:212:in `call'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/listen-0.7.3/lib/listen/listener.rb:212:in `on_change'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/listen-0.7.3/lib/listen/listener.rb:221:in `block in initialize_adapter'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/listen-0.7.3/lib/listen/adapter.rb:198:in `call'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/listen-0.7.3/lib/listen/adapter.rb:198:in `report_changes'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/listen-0.7.3/lib/listen/adapter.rb:210:in `poll_changed_dirs'
> [#] /Users/h/.rvm/gems/ruby-1.9.3-p194/gems/listen-0.7.3/lib/listen/adapters/darwin.rb:38:in `block in start'
```

**Comment by joechrysler** [comment]
I'm getting this one too.

**Comment by aaronmcadam** [comment]
Looks like https://github.com/hawx/guard-sass/issues/41 was opened to look at this again.

**Comment by rymai** [comment]
Hi guys, it's because there wasn't any new version released to Rubygems in the past 2.5 years!!

I've now released 0.0.8! Please re-open the issue if needed, thanks!
